### PR TITLE
Adding loading to search bar and refactor of redirect logic

### PR DESCRIPTION
**Pull request by tejans24**
connects #3059 #2342 #2947 

screen shot 2017-08-28 at 9 52 34 am

Known issues: 
   * The loading button doesn't reset state when error happens. https://github.com/department-of-veterans-affairs/caseflow/pull/3089 takes care of this.

AC:
* When searching for a query, the search bar button will change state to loading
* This PR also refactored the redirect logic when appeals are fetched from the server
    *  When DEMO456 is used as a search term, user should be redirected to the proper claims folder
    * When DEMO123 is used as a search term, user should be shown multiple appeals to select from  in a Modal.

### CODE REVIEW

File: client/app/reader/reducer.js [hunk 1]
```diff
@@ -797,10 +798,27 @@ export const reducer = (state = initialState, action = {}) => {
         }
       }
     });
-  case Constants.RECEIVE_MULTIPLE_APPEALS_USING_VETERAN_ID:
+  case Constants.REQUEST_APPEAL_USING_VETERAN_ID:
     return update(state, {
       ui: {
         caseSelect: {
+          requestAppealsUsingVeteranId: { $set: true }
```
Comment (mdbenjam) [inline client/app/reader/reducer.js]: can we change this variable name to something that sounds like a boolean. Like isRquestingAppealUsingVeteranId? It sounds more like an action right now.

File: client/app/reader/actions.js [hunk 2]
```diff
@@ -627,29 +627,24 @@ export const fetchAppealUsingVeteranIdFailed = () => ({
...
+});
+
 export const fetchAppealUsingVeteranId = (veteranId) => (
   (dispatch) => {
+    dispatch(requestAppealUsingVeteranId());
     ApiUtil.get(`/reader/appeal/veteran-id/${veteranId}?json`).then((response) => {
       const returnedObject = JSON.parse(response.text);
-      const numOfAppeals = _.size(returnedObject.appeals);
 
       // if the veteran only has one appeal associated,
```
Comment (mdbenjam) [inline client/app/reader/actions.js]: Comment seems not relevant any more.

File: client/app/reader/reducer.js [hunk 3]
```diff
@@ -797,10 +798,27 @@ export const reducer = (state = initialState, action = {}) => {
...
+        }
+      }
+    });
+  case Constants.RECEIVE_APPEALS_USING_VETERAN_ID_FAILED:
```
Comment (mdbenjam) [inline client/app/reader/reducer.js]: Can we change this to be `_FAILURE` for consistency with other actions?

File: client/app/reader/reducer.js [hunk 4]
```diff
@@ -797,10 +798,27 @@ export const reducer = (state = initialState, action = {}) => {
...
+    return update(state, {
+      ui: {
+        caseSelect: {
+          requestAppealsUsingVeteranId: { $set: false }
+        }
+      }
+    });
+  case Constants.RECEIVE_APPEALS_USING_VETERAN_ID:
```
Comment (mdbenjam) [inline client/app/reader/reducer.js]: Can we keep this constant to end with `_SUCCESS`

File: client/app/reader/actions.js [hunk 5]
```diff
@@ -627,29 +627,24 @@ export const fetchAppealUsingVeteranIdFailed = () => ({
...
 });
 
+export const requestAppealUsingVeteranId = () => ({
+  type: Constants.REQUEST_APPEAL_USING_VETERAN_ID
+});
+
 export const fetchAppealUsingVeteranId = (veteranId) => (
   (dispatch) => {
+    dispatch(requestAppealUsingVeteranId());
     ApiUtil.get(`/reader/appeal/veteran-id/${veteranId}?json`).then((response) => {
```
Comment (mdbenjam) [inline client/app/reader/actions.js]: You don't have a catch here. We need to call the failure action.

Comment (tejans24) [reply client/app/reader/actions.js]: This error handling is in this PR: https://github.com/department-of-veterans-affairs/caseflow/pull/3089 which handles the error state.

Comment (mdbenjam) [reply client/app/reader/actions.js]: I see this is being done in another PR.

File: client/app/reader/reducer.js (hunk 3 above)
Comment (tejans24) [reply client/app/reader/reducer.js]: Good point

**Pull request merged**